### PR TITLE
handle decimals in formatAmount, also add locale to swap widget

### PR DESCRIPTION
**Pull request by inclinedmind**
Update formatAmount to handle tokens with non standard decimals when displaying values in the swap widget.

Adds locale to swap widget. I guess we should update all formatAmount calls throughout the app to include the locale too.

Also updates formatAmount, formatCompactAmount and formatPercent to take in an options object.

Screen Shot 2023-08-11 at 4 17 55 PM
Screen Shot 2023-08-11 at 4 18 01 PM

### CODE REVIEW

File: packages/utils/src/formatValue.ts [hunk 1]
```diff
@@ -41,9 +41,17 @@ function formatUnitPercent(paramLocale?: string) {
   });
 }
 
-export function formatAmount(amount: bigint, locale?: string, unit: 'wad' | 'ray' = 'wad'): `${number}` {
-  const amountToFormat = unit === 'ray' ? formatRay(amount) : formatWad(amount);
-
+export function formatAmount(
+  amount: bigint,
```
Comment (codewiththeflowxyz) [inline packages/utils/src/formatValue.ts:45]: Now that the function arguments are growing it makes sense to leave it as 

`formatAmount(amount: bigint, options: {locale?: string, unit: 'wad' | 'ray' | number = 'wad'})`

**Review by codewiththeflowxyz — APPROVED** [review]
LGTM, just one suggestion, wdyt?

File: packages/utils/src/formatValue.ts (hunk 1 above)
Comment (inclinedmind) [reply packages/utils/src/formatValue.ts:45]: yeah that's probably easier than dealing with two optional arguments, I'll update

**Review by iceanddust — APPROVED** [review]
LGTM

**Pull request merged**